src/Controller/Api/TreeController.php

```diff
@@ -278,11 +284,19 @@ public function getLettersAction(Request $request)
 
         $firstLetterLastName = $qb->expr()->upper($qb->expr()->substring('person.lastName', 1, 1));
 
-        $query = $qb
+        $qb
             ->select($firstLetterLastName)
             ->distinct()
-            ->orderBy($firstLetterLastName)
-            ->getQuery();
+            ->orderBy($firstLetterLastName);
+
+        if ($fundingOrgFilter->isActive()) {
+            $qb->innerJoin('person.personResearchGroups', 'prg');
+            $qb->innerJoin('prg.researchGroup', 'rg');
+            $qb->where('rg.id IN (:fos)');
+            $qb->setParameter('fos', $fundingOrgFilter->getResearchGroupsIdArray());
```

Comment:
Instead of using this triple nested foreach method to get research group ids, we can just use two more inner joins to get the result. Something like this
`            $qb->innerJoin('person.personResearchGroups', 'prg');
            $qb->innerJoin('prg.researchGroup', 'rg');
            $qb->innerJoin('rg.fundingCycle', 'fundCyc');
            $qb->innerJoin('fundCyc.fundingOrganization','fundOrg');
            $qb->andWhere('fundOrg.id IN (:fundingOrgIds)');
            $qb->setParameter('fundingOrgIds', $fundingOrgFilter->getFilterIdArray());`

Reply:
Why not go straight to shortnames?  We may have to this 2 or 3 times, and the list of researchgroups is already available from the utility, which only had to run the query once. It also a little more efficient, but it's more about decentralizing in my mind.

Reply:
yes, we can use funding org short names too instead of Ids. But i think using inner joins to filter is more efficient that using the getResearchGroupsIdArray() method